SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/GameModeReplicationTest/GameModeReplicationTest.cpp

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "GameModeReplicationTest.h"
+
+#include "Net/UnrealNetwork.h"
+
+#include "GameFramework/GameStateBase.h"
+
+AGameModeReplicationTestGameMode::AGameModeReplicationTestGameMode()
+{
+	NetCullDistanceSquared = 0;
+}
+
+void AGameModeReplicationTestGameMode::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	DOREPLIFETIME(AGameModeReplicationTestGameMode, ReplicatedValue);
+}
+
+bool AGameModeReplicationTestGameMode::IsNetRelevantFor(const AActor* RealViewer, const AActor* ViewTarget,
+														const FVector& SrcLocation) const
+{
+	return false;
+}
+
+/**
+ * This test checks that GameModes:
+ * * Only owned by a single worker
+ * * Replicate between servers
+ * * Don't replicate to clients
+ */
+
+AGameModeReplicationTest::AGameModeReplicationTest()
+{
+	Author = TEXT("Dmitrii");
+	Description = TEXT("Test GameMode replication");
+
+	bReplicates = true;
+	bAlwaysRelevant = true;
+}
+
+void AGameModeReplicationTest::MarkWorkerGameModeAuthority_Implementation(int WorkerId, bool bHasGameModeAuthority)
+{
+	ServerResponsesCount++;
+
+	if (bHasGameModeAuthority)
+	{
+		++AuthorityServersCount;
+	}
+}
+
+void AGameModeReplicationTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	check(GetWorld()->GetGameState()->GameModeClass == AGameModeReplicationTestGameMode::StaticClass());
+
+	AuthorityServersCount = 0;
+
+	AddStep(TEXT("Changing replicated value on the authoritative server"), FWorkerDefinition::AllServers, nullptr, [this]() {
+		AGameModeReplicationTestGameMode* GameMode = Cast<AGameModeReplicationTestGameMode>(GetWorld()->GetAuthGameMode());
+
+		check(IsValid(GameMode));
+
+		AssertEqual_Int(GameMode->ReplicatedValue, AGameModeReplicationTestGameMode::StartingValue,
+						TEXT("Value on the GameMode before changing it"));
+
+		const bool bHasAuthorityOverGameMode = GameMode->HasAuthority();
+
+		MarkWorkerGameModeAuthority(GetLocalWorkerId(), bHasAuthorityOverGameMode);
+
+		if (bHasAuthorityOverGameMode)
+		{
+			const FVector GameModeFarawayLocation(1000 * 1000 * 1000, 1000 * 1000 * 1000, 1000 * 1000 * 1000);
+
+			// GameModeBase hides this function so we upcast to AActor;
+			// this probably doesn't do anything as GameModes generally don't have RootComponents
+			// and no location as consequence
+			AActor* GameModeActor = GameMode;
+
+			GameModeActor->SetActorLocation(GameModeFarawayLocation);
```

Comment:
Why are we moving the game mode?

Reply:
Ah I'm guessing it's so it goes out of interest? I wonder if we can instead set up the LB strategy to be a 1x3 grid? (so 2 of the servers don't check out the middle area)
My concern is simply that moving the game mode is not something that would normally happen in Unreal.

Reply:
Yeah moving it outside of other servers' interest area was the intent, as suggested by @m-samiec; however, GameModes don't have a RootComponent so changing its position fails anyway (unless Epic change this in a future version!).

This doesn't make any difference in the context of the test as LB strategy in the TestGym map is setup to have 3x3 grid (1x3 would probably work too with smaller server worker count, so will check this too!)

Reply:
1x3 works perfectly, and I think I'll delete this location change.